toolchain/check/import_ref.cpp

```diff
@@ -1359,19 +1383,22 @@ class ImportRefResolver {
   auto TryResolveTypedInst(SemIR::BaseDecl inst, SemIR::InstId import_inst_id)
       -> ResolveResult {
     auto type_const_id = GetLocalConstantId(inst.type_id);
-    auto base_type_const_id = GetLocalConstantId(inst.base_type_id);
+    auto base_type_const_id = GetLocalConstantId(
+        import_ir_.constant_values().Get(inst.base_type_inst_id));
     if (HasNewWork()) {
       return Retry();
     }
 
-    // Import the instruction in order to update contained base_type_id and
-    // track the import location.
+    auto base_type_inst_id = AddLoadedImportRef(
+        context_, {.ir_id = import_ir_id_, .inst_id = inst.base_type_inst_id},
+        SemIR::TypeId::TypeType, base_type_const_id);
```

Comment:
Is it worth trying to share logic with `AdaptDecl`, maybe with a template? It looks like this section mirrors.

Of course, maybe the answer is just "that's not how we do it elsewhere" or "think about it as part of the other refactoring"

Reply:
Let's think about it as part of the other refactoring :-)